src/test/java/uk/gov/pay/api/utils/ConnectorMockClient.java

```diff
@@ -60,11 +63,14 @@ private String createChargeResponse(long amount, String chargeId, String status,
                 .add("return_url", returnUrl)
                 .add("payment_provider", paymentProvider)
                 .add("created_date", createdDate)
-                .add("links", asList(links))
-                .build();
+                .add("links", asList(links));
+        if (gatewayTransactionId != null) {
+            jsonStringBuilder.add("gateway_transaction_id", gatewayTransactionId);
+        }
```

Comment:
Indeed, you see that because effectively Connector is sending it back but we are ignoring it and It wasn't reflected in the tests so I created a failing test and making sure we don't map this field.